pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go

```diff
@@ -554,14 +554,29 @@ func handleRangefeedError(ctx context.Context, err error) (rangefeedErrorInfo, e
 	}
 }
 
+// catchup alloc is a catchup scan allocation.
+type catchupAlloc func()
```

Comment:
general nit: This commit is a mix of 2 fixes - the counter leak and the context. I'd prefer them to be in separate commits or PRs, so that the 2 fixes could be reviewed separately, and (if ever needed) rolled-back/cherry-picked/backported on individual basis.

Reply:
Split.